src/extension/executors/utils.ts

```diff
@@ -46,21 +98,18 @@ export async function retrieveShellCommand (exec: NotebookCellExecution) {
   let cellText = exec.cell.document.getText()
   const cwd = path.dirname(exec.cell.document.uri.fsPath)
   const rawText = exec.cell.document.getText()
-  const exportMatches: string[] = (rawText.endsWith('\n') ? rawText : `${rawText}\n`)
-    .match(EXPORT_EXTRACT_REGEX) || []
+
+  const exportMatches = getCommandExportExtractMatches(rawText)
 
   const stateEnv = Object.fromEntries(ENV_STORE)
-  for (const e of exportMatches) {
-    const [key, ph] = e.trim().slice('export '.length).split('=')
-    const hasStringValue = ph.startsWith('"') || ph.startsWith('\'')
-    const placeHolder = hasStringValue ? ph.slice(1, -1) : ph
 
-    if (placeHolder.startsWith('$(') && placeHolder.endsWith(')')) {
+
+  for (const { hasStringValue, key, match, type, value } of exportMatches) {
+    if (type === 'exec') {
       /**
        * evaluate expression
        */
-      const expression = placeHolder.slice(2, -1)
-      const expressionProcess = cp.spawn(expression, {
+      const expressionProcess = cp.spawn(value, {
```

Comment:
Shouldn't this use the Runner APIs versus spawning child processes? Consider addressing in follow up issue/PR.

Reply:
This feature uses `getCommandExportExtractMatches` directly - it does not call `retrieveShellCommand`.

Reply:
great 👌 , thanks for clarifying!